src/mango/src/mango_idx.hrl

```diff
@@ -10,6 +10,8 @@
 % License for the specific language governing permissions and limitations under
 % the License.
 
+-ifndef(MANGO_IDX_DEFINED).
+-define(MANGO_IDX_DEFINED, 1).
```

Comment:
Wonder if there is a way to avoid this `ifndef $already_defined` pattern, it's not a very common in CouchDB code at least.

See if it's possible to move the `idx` record to `mango.hrl` and just have everyone include that. In most case when we include mango_idx.hrl we include mango.hrl as well. One place were we don't is mango_native_proc.erl but I don't think we're even using mango_idx.hrl there at all.

Reply:
I was also a bit hesitant to employ this trick, but I left it there at the end.  I reworked the solution as you suggested and it seems to be working.  Thanks!